sql/core/src/test/scala/org/apache/spark/sql/CollationSQLExpressionsSuite.scala

```diff
@@ -1151,6 +1151,23 @@ class CollationSQLExpressionsSuite
     })
   }
 
+  test("Support InputFileName expression with collation") {
+    // Supported collations
+    Seq("UTF8_BINARY", "UTF8_BINARY_LCASE", "UNICODE", "UNICODE_CI").foreach(collationName => {
```

Comment:
Stopping by this PR, maybe this was discussed a while ago, just curious: why these possible `collationName` is not listed by a enum?

Reply:
I'd say most of the time tests are custom, depending on the function, and only rarely do we have functions with no parameters - so this is not very common (actually, this may be the only such function in this test suite). So I'd say there's no need for a preset suite-level collation name collation.

On the other hand, we will soon have hundreds/thousands of collations in Spark, so these 4 collations will soon appear to be very random (i.e. we won't run this test for **all** supported collations, but these 4 should still be enough to cover all cases for all functions).